services/datalad/datalad_service/handlers/upload.py

```diff
@@ -38,7 +37,6 @@ def _finish_upload(self, dataset_id, upload, name, email, cookies):
                 gevent.sleep()
                 move_files(upload_path, ds.path)
                 ds.save(unlock_files)
-                update_head(ds, dataset_id, cookies)
```

Comment:
This is not the right approach for preventing snapshots on upload, this will still create a snapshot on the next commit instead of only when the user creates one.

Here's where that actually happens:
https://github.com/OpenNeuroOrg/openneuro/blob/0fff635c53753cd1c49c4eafc8c30f4ebdcb13fe/packages/openneuro-server/src/graphql/resolvers/draft.js#L24-L36

It might be possible to drop updateRef entirely as this is the only thing it is responsible for at the moment.

Reply:
Opted to just remove the section where a new snapshot is created instead of dropping updateRef entirely.